### PR TITLE
feat: remove duplicated schema covert from arrow schema

### PR DESCRIPTION
**Pull request by Rachelint**
## Related Issues
Closes #

## Detailed Changes
+ Add a `RecordBatchBuilder` which will cache the schema coverted from arrow's.
+ Build inner `RecordBatch` using `RecordBatchBuilder`.

## Test Plan 
Test by exist tests.

### CODE REVIEW

File: common_types/src/record_batch.rs [hunk 1]
```diff
@@ -269,19 +269,30 @@ impl RecordBatch {
...
+    schema_cache: Option<RecordSchema>,
+}
 
-    fn try_from(arrow_record_batch: ArrowRecordBatch) -> Result<Self> {
-        let record_schema =
-            RecordSchema::try_from(arrow_record_batch.schema()).context(ConvertArrowSchema)?;
+impl RecordBatchBuilder {
+    pub fn build(&mut self, arrow_record_batch: ArrowRecordBatch) -> Result<RecordBatch> {
+        let schema = match &self.schema_cache {
+            Some(schema) => schema.clone(),
```
Comment (chunshao90) [inline common_types/src/record_batch.rs:280]: Is it possible that the schema is not consistent?

File: common_types/src/record_batch.rs [hunk 2]
```diff
@@ -269,19 +269,30 @@ impl RecordBatch {
...
+#[derive(Default)]
+pub struct RecordBatchBuilder {
+    schema_cache: Option<RecordSchema>,
+}
 
-    fn try_from(arrow_record_batch: ArrowRecordBatch) -> Result<Self> {
-        let record_schema =
-            RecordSchema::try_from(arrow_record_batch.schema()).context(ConvertArrowSchema)?;
+impl RecordBatchBuilder {
+    pub fn build(&mut self, arrow_record_batch: ArrowRecordBatch) -> Result<RecordBatch> {
```
Comment (jiacai2050) [inline common_types/src/record_batch.rs:278]: It's wired to name this `build`, usually build is only called once.

`convert` maybe a better name.

File: common_types/src/record_batch.rs (hunk 1 above)
Comment (Rachelint) [reply common_types/src/record_batch.rs:280]: Now,  such usecase not exist.
But it is needed to be considered actually, I will add some comments about it.

Comment (Rachelint) [reply common_types/src/record_batch.rs:288]: Some comments about are added.

File: common_types/src/record_batch.rs (hunk 2 above)
Comment (Rachelint) [reply common_types/src/record_batch.rs]: Done.

File: common_types/src/record_batch.rs [hunk 3]
```diff
@@ -269,27 +269,63 @@ impl RecordBatch {
...
+///
+/// Converting arrow schema to inner's is expensive. For eliminate the
+/// duplicated converting, we define such converter to cache the first converted
+/// one and reuse it in later converting.
+///
+/// NOTE: must ensure the converted arrow record batches has the same schema
+/// before deciding to use this converter.
+#[derive(Default)]
+pub struct CachedRecordBatchesConverter {
+    schema_cache: Option<RecordSchema>,
```
Comment (jiacai2050) [inline common_types/src/record_batch.rs:282]: How about init this field in `fn new()`, then `convert` could take `&self` directly.

**Pull request closed without merging**